src/components/module-list/module-list.component.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from "react";
+import "./module-list.styles.css";
+import { Card } from "../module/module.component";
+import Submit from "../submit/submit.component";
+
+const { db, auth } = require("../../firebase/index.firebase");
+const axios = require("axios");
+
+const ModuleList = () => {
+  const [nusmodules, setNusmodules] = useState([]);
+
+  const addModules = async (nusmodules) => {
+    let modules = nusmodules.split("&");
+    let modulesTaken = [];
+    let mod;
+    for (mod of modules) {
+      if (mod.includes("?")) {
+        mod = mod.split("?");
+        mod = mod[1];
+      }
+      mod = mod.split("=", 1);
+      await axios
+        .get(`https://api.nusmods.com/v2/2019-2020/modules/${mod}.json`)
+        .then(function (response) {
+          if (response.status === 200) {
+            modulesTaken.push(mod[0]);
+          }
+        })
+        .catch(function (error) {
+          console.log(error);
+        });
+    }
+
+    let moduleFields = {
+      mods_taken: modulesTaken,
+    };
+
+    var user = auth().currentUser;
```

Comment:
I tried this one a while ago and it worked.
But recently it stopped working this way, so I assigned Auth = firebase.auth(), then assign user = Auth.currentUser.
It has worked since then, so I will update this soon

Reply:
auth is indeed = firebase.auth() it is defined in the index.firebase.js, it is supposed to be user=auth.currentUser

Reply:
I understand. 
Previously I got an error when I did this, but now it goes well